pkg/kubelet/status/status_manager.go

```diff
@@ -879,9 +887,26 @@ func mergePodStatus(oldPodStatus, newPodStatus v1.PodStatus, couldHaveRunningCon
 			podConditions = append(podConditions, c)
 		}
 	}
+
+	transitioningToTerminalPhase := !podutil.IsPodPhaseTerminal(oldPodStatus.Phase) && podutil.IsPodPhaseTerminal(newPodStatus.Phase)
+
 	for _, c := range newPodStatus.Conditions {
 		if kubetypes.PodConditionByKubelet(c.Type) {
 			podConditions = append(podConditions, c)
+		} else if kubetypes.PodConditionSharedByKubelet(c.Type) {
+			if c.Type == v1.AlphaNoCompatGuaranteeDisruptionTarget {
+				// update the pod disruption condition only if transitioning to terminal phase. In particular, check if
+				// there might still be running containers to avoid sending an unnecessary PATCH request if the
+				// actual transition is delayed (see below)
+				if transitioningToTerminalPhase && !couldHaveRunningContainers {
+					// update the LastTransitionTime
```

Comment:
nit: note that the reason we need to update `LastTransitionTime` for this condition again here is because the older transition time set in ` updateStatusInternal` is likely stale because the status update was not sent until the all the containers have terminated.